tests/unit/helm/api-deployment_test.go

```diff
@@ -217,6 +217,165 @@ func (s *deploymentApiTemplateTest) TestReplicas() {
 	}
 }
 
+func (s *deploymentApiTemplateTest) TestTopologySpreadConstraints() {
+	testCases := []struct {
+		name     string
+		values   map[string]string
+		expected func(constraint []corev1.TopologySpreadConstraint)
+	}{
+		{
+			"defaultValues",
+			nil,
+			func(constraint []corev1.TopologySpreadConstraint) {
+				var expectedTopologySpreadConstraint []corev1.TopologySpreadConstraint
+				s.Equal(expectedTopologySpreadConstraint, constraint, "Constraints should be equal")
+			},
+		},
+		{
+			"overrideTopologySpreadConstraintsRequiredValues",
+			map[string]string{
+				"apiSettings.topologySpreadConstraints[0].maxSkew":           "1",
+				"apiSettings.topologySpreadConstraints[0].topologyKey":       "kubernetes.io/hostname",
+				"apiSettings.topologySpreadConstraints[0].whenUnsatisfiable": "DoNotSchedule",
+			},
+			func(constraint []corev1.TopologySpreadConstraint) {
+				var expectedTopologySpreadConstraint []corev1.TopologySpreadConstraint
+				eexpectedTopologySpreadConstraintJSON := `[
```

Comment:
`eexpectedTopologySpreadConstraintJSON` starts with `ee`. Is that expected?

Reply:
Fixed typo